heltour/tournament/views.py

```diff
@@ -1345,6 +1345,59 @@ def team(season):
                 season_score += bye.score()
                 season_score_total += 1
         season_perf_rating = season_perf.calculate()
+        
+        def season_performance(season):
+            season_score = 0
+            season_score_total = 0
+            season_perf = PerfRatingCalc()
+
+            games = defaultdict(list)
+            if season is None:
+                byes = {}
+            elif season.league.competitor_type == 'team':
+                pairings = TeamPlayerPairing.objects.filter(white=player) | TeamPlayerPairing.objects.filter(black=player)
+                for p in pairings.filter(team_pairing__round__season=season).order_by('team_pairing__round__number').nocache():
+                    games[p.team_pairing.round.number].append(p)
+                byes = {}
+            else:
+                pairings = LonePlayerPairing.objects.filter(white=player) | LonePlayerPairing.objects.filter(black=player)
+                for p in pairings.filter(round__season=season).order_by('round__number').nocache():
+                    games[p.round.number].append(p)
+                byes = {b.round.number: b for b in PlayerBye.objects.filter(round__season=season, player=player)}
+
+            for round_ in season.round_set.filter(publish_pairings=True).order_by('number'):
+                if round_.number in games:
+                    for p in games[round_.number]:
+                        if p.result == '':
+                            continue
+                        game_score = p.white_score() if p.white == player else p.black_score()
+                        if game_score is not None:
+                            season_score += game_score
+                            season_score_total += 1
+                        # Add pairing to performance calculation
+                        if p.game_played() and p.white is not None and p.black is not None:
+                            sp = SeasonPlayer.objects.filter(season=season, player=p.black if p.white == player else p.white).first()
+                            if sp is not None and sp.seed_rating is not None:
+                                opp_rating = sp.seed_rating
+                            else:
+                                opp_rating = p.black_rating_display() if p.white == player else p.white_rating_display(self.league)
+                            season_perf.add_game(game_score, opp_rating)
+                elif round_.number in byes:
+                    bye = byes[round_.number]
+                    season_score += bye.score()
+                    season_score_total += 1
+            return season_perf
+
+        def league_performance(seasons):
+            league_perf_obj = PerfRatingCalc()
+            for season in seasons:
+                league_perf_obj.merge(season_performance(season[0]))
+            return league_perf_obj
+
+        for leaguetype in other_season_leagues:
```

Comment:
In general, prefer deconstruction syntax rather than indexing into a tuple. For example,
`for league, season_info in other_season_leagues`

In this case, since you're not using most of that (convoluted) type anyway, you should just query for the data you want:
`[sp.season for sp in player.seasonplayer_set.filter(season__league=self.league)]`
And then you can get rid of this loop entirely, and make it clearer what data you are using in `league_performance`.